src/lyra/semantics/semantics.py

```diff
@@ -90,9 +93,81 @@ def list_display_access_semantics(self, stmt: ListDisplayAccess, state: State) -
         """
         items = [self.semantics(item, state).result for item in stmt.items]
         result = set()
-        for combination in itertools.product(*items):
-            display = ListDisplay(ListLyraType(combination[0].typ), list(combination))
-            result.add(display)
+        if items:   # not empty
+            for combination in itertools.product(*items):
+                display = ListDisplay(ListLyraType(combination[0].typ), list(combination))
+                result.add(display)
+        else:
+            result.add(ListDisplay(ListLyraType(LyraType())))     # empty list of generic type
```

Comment:
Ah, this is why you made the `__repr__` method non-abstract... Maybe add an AnyLyraType instead. 

I guess at some point I should modify things such that the type comes from the type annotations in the program.